lib/Doctrine/ORM/Proxy/ProxyFactory.php

```diff
@@ -191,14 +164,13 @@ public function __construct(
     }
 
     /**
-     * {@inheritDoc}
+     * @param class-string $className
+     * @param array<mixed> $identifier
+     *
+     * @return InternalProxy
```

Comment:
Using a native return type here will involve modify some tests.

Reply:
I think we should fix those tests, but that could be done in a follow-up.